### PR TITLE
CI setup

### PR DESCRIPTION
**Pull request by come-nc**
Make sure CI is green, add missing configurations and github workflows.

### CODE REVIEW

**Comment by come-nc** [comment]
All good apart from stylelint which I do not know how to fix.

File: .github/workflows/command-compile.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# SPDX-FileCopyrightText: Nextcloud contributors
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+name: Compile Command
```
Comment (susnux) [inline .github/workflows/command-compile.yml:4]: I do not think we need this, we do not store assets (hopefully)

File: .github/workflows/lint-php-cs.yml [hunk 2]
```diff
@@ -22,17 +24,19 @@ jobs:
...
-        uses: actions/checkout@v2
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
 
-      - name: Set up php ${{ matrix.php-versions }}
-        uses: shivammathur/setup-php@v2
+      - name: Set up php
+        uses: shivammathur/setup-php@4bd44f22a98a19e0950cbad5f31095157cc9621b # v2
         with:
-          php-version: "7.4"
+          php-version: 8.1
```
Comment (susnux) [inline .github/workflows/lint-php-cs.yml:32]: Should we better use minimum supported 8.0 or max. supported 8.3?

File: .github/workflows/lint-php.yml [hunk 3]
```diff
@@ -5,38 +5,51 @@
...
+  group: lint-php-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
 jobs:
   php-lint:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        php-versions: ["7.4", "8.0", "8.1"]
+        php-versions: [ "8.0", "8.1", "8.2" ]
```
Comment (susnux) [inline .github/workflows/lint-php.yml:30]: Same as above, probably add 8.3 directly

File: .github/workflows/phpunit-mysql.yml [hunk 4]
```diff
@@ -1,32 +1,50 @@
...
+  cancel-in-progress: true
 
 jobs:
   phpunit-mysql:
     runs-on: ubuntu-latest
 
     strategy:
       matrix:
-        php-versions: ['7.4', '8.0', '8.1']
+        php-versions: ['8.0', '8.1', '8.2']
```
Comment (susnux) [inline .github/workflows/phpunit-mysql.yml:42]: Same as above including 8.3?

Also I would like to have this file renamed to phpunit-mariadb as this is not mysql.

File: .github/workflows/psalm.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    name: Nextcloud
+    steps:
+      - name: Checkout
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      - name: Set up php
+        uses: shivammathur/setup-php@4bd44f22a98a19e0950cbad5f31095157cc9621b # v2
+        with:
+          php-version: 8.1
```
Comment (susnux) [inline .github/workflows/psalm.yml:34]: any reason for 8.1? Why not min. 8.0 or max 8.3? Or recommended 8.2?^^

File: composer.json [hunk 6]
```diff
@@ -8,30 +8,35 @@
...
 			"composer/package-versions-deprecated": true
 		},
 		"platform": {
-			"php": "8.1"
+			"php": "8.0.2"
 		}
 	},
-	"require": {
+	"require-dev": {
+		"phpunit/phpunit": "^9",
```
Comment (susnux) [inline composer.json:37]: 10 does not work?

**Review by susnux — COMMENTED** [review]
Some comments :see_no_evil:

File: .github/workflows/lint-php.yml (hunk 3 above)
Comment (come-nc) [reply .github/workflows/lint-php.yml:30]: Same, comes from https://github.com/nextcloud/.github should be added there.

File: .github/workflows/phpunit-mysql.yml (hunk 4 above)
Comment (come-nc) [reply .github/workflows/phpunit-mysql.yml:42]: Open a PR on https://github.com/nextcloud/.github :-P

File: .github/workflows/psalm.yml (hunk 5 above)
Comment (come-nc) [reply .github/workflows/psalm.yml:34]: My guess is it was the minimum for a recent psalm at some point

Comment (come-nc) [reply .github/workflows/psalm.yml:34]: https://github.com/nextcloud/.github/pull/99 :shrug:

File: composer.json (hunk 6 above)
Comment (come-nc) [reply composer.json:37]: Not on 8.0

Comment (susnux) [reply composer.json:37]: sad then we have migrate some day later while we could start with the new stuff on the beginning :(

**Pull request merged**